packages/main/src/Menu.ts

```diff
@@ -190,6 +190,32 @@ type OpenerStandardListItem = StandardListItem & { associatedItem: MenuItem };
  * @since 1.10.0
  */
 @event("after-close")
+
+/**
+ * Fired when a menu item receives focus.
+ *
+ * @public
+ * @param { HTMLElement } ref The currently focused element representing a <code>ui5-menu-item</code>.
+ * @param { HTMLElement } item The <code>ui5-menu-item</code> represented by the focused element.
+ * @since 1.23.1
+ */
+@event("item-focus", {
```

Comment:
Every public event with details should be described with type used in the event decorator @event<TypeForDetails>("item-focus" and exported.

In short:

- add type describing the event details
`type MenuItemFocusEventDetail = {
         ref: HTMLElement
		 item:  HTMLElement,
}`

-  then use it in the event decorator
 `@event<MenuItemFocusEventDetail>("item-focus",`
 
 - and use it when firing the event
` this.fireEvent<MenuItemFocusEventDetail>("item-focus",`
 
 - Finally export it:
 `export type {
    MenuItemFocusEventDetail,
 }`
 
 And, maybe we can see if "ref" or targetRef is better as I think similar event detail param is called targetRef in other places.

Reply:
I agree with the comment for targetRef. But I also see that we are using it in Tokenizer and MutliComboBox, should we think in direction of unifying it in all components?

Reply:
I'd suggest we keep the current name (`ref`) as this is what the client is using from the PoC and then unify them all for 2.0